### PR TITLE
[----------------- rover initialized]

### PR DESCRIPTION
**Pull request by leogrim**
@eishay @stkem @andrewconner @yingjieMiao @2is10

### CODE REVIEW

**Comment by yingjieMiao** [comment]
:+1:

**Comment by stkem** [comment]
Great! (but soooooooooo much boilerplate :-/)

**Comment by leogrim** [comment]
There's some non mandatory stuff though, e.g. a default service controller.
But I meant that PR to be a tutorial of sorts.

File: kifi-backend/modules/rover/app/com/keepit/rover/common/cache/RoverCacheModule.scala [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+import com.keepit.graph.model._
+import com.keepit.model._
+import com.keepit.model.cache.UserSessionViewExternalIdCache
+import com.keepit.social.BasicUserUserIdCache
+import com.keepit.search.{ ArticleSearchResultCache, InitialSearchIdCache, ActiveExperimentsCache }
+import com.keepit.common.usersegment.UserSegmentCache
+import scala.concurrent.duration._
+import com.keepit.common.cache._
+
+case class RoverCacheModule(cachePluginModules: CachePluginModule*) extends CacheModule(cachePluginModules: _*) {
```
Comment (andrewconner) [inline kifi-backend/modules/rover/app/com/keepit/rover/common/cache/RoverCacheModule.scala:15]: Biggest area for improvement. This shouldn't be necessary :(

**Comment by leogrim** [comment]
For future reference, couple of amendments with this PR:
- https://github.com/kifi/keepit/pull/13966
- https://github.com/kifi/keepit/pull/13960

**Pull request merged**